test/hbbft_SUITE.erl

```diff
@@ -343,6 +347,91 @@ one_actor_corrupted_key_test(Config) ->
     ct:log("chain contains ~p distinct transactions~n", [length(BlockTxns)]),
     ok.
 
+one_actor_oversized_batch_test(Config) ->
+    N = proplists:get_value(n, Config),
+    F = proplists:get_value(f, Config),
+    B = proplists:get_value(batchsize, Config),
+    PK = proplists:get_value(pubkey, Config),
+    [SK1 | SKs] = proplists:get_value(privatekeys, Config),
+
+    BadWorker = % with an oversized batch parameter
+        ok(hbbft_worker:start_link(N, F, 0, tpke_privkey:serialize(SK1), B + 1, false)),
+    GoodWorkers =
+        [ok(hbbft_worker:start_link(N, F, I + 1, tpke_privkey:serialize(SKi), B, false))
+        || {I, SKi} <- enumerate(SKs)
+        ],
+
+    % Each node needs at least B/N transactions.
+    GoodTxns = [list_to_binary("GOOD_" ++ integer_to_list(I)) || I <- lists:seq(1, B * N)],
+    BadTxns  = [list_to_binary("BAD_"  ++ integer_to_list(I)) || I <- lists:seq(1, B)],
+
+    % Submit transactions:
+    lists:foreach(
+      fun (T) -> ok = hbbft_worker:submit_transaction(T, BadWorker) end,
+      BadTxns),
+    lists:foreach(
+      fun (T) ->
+              Destinations = random_n(rand:uniform(N), GoodWorkers),
+              [ok = hbbft_worker:submit_transaction(T, D) || D <- Destinations]
+      end,
+      GoodTxns),
+
+    % Wait for all the worker's mailboxes to settle:
+    ok = wait_for_chains([BadWorker | GoodWorkers], 2),
+
+    % Wait for the chains to converge:
+    Chains = get_common_chain(GoodWorkers),
+    1 = sets:size(Chains),
+    [Chain] = sets:to_list(Chains),
+    CommittedTxns =
+        lists:flatten([hbbft_worker:block_transactions(Block) || Block <- Chain]),
+
+    % Assert transactions are cryptographically linked:
+    true = hbbft_worker:verify_chain(Chain, PK),
+
+    % Assert transactions are unique:
+    [] = CommittedTxns -- lists:usort(CommittedTxns),
```

Comment:
please use `?assertEqual()` here and two lines below for better quality printing.